### PR TITLE
fix(connlib): Fix `getSystemDefaultResolvers` while tunnel session is active

### LINKED ISSUE #3235: Cache DNS resolvers App-side so that `getSystemDefaultResolvers` works during portal reconnect for non-Apple platforms
https://firezonehq.slack.com/archives/C067DSY7TFX/p1705337844386619?thread_ts=1705336174.619669&cid=C067DSY7TFX

### PR DESCRIPTION
**Pull request by jamilbk**
Reverts firezone/firezone#3198

This was actually functioning well for non-Apple platforms. Marking as a draft until #3235 PR is opened

Fixes #3235

### CODE REVIEW

File: swift/apple/FirezoneNetworkExtension/Adapter.swift [hunk 1]
```diff
@@ -141,6 +141,12 @@ class Adapter {
 
       self.logger.log("Adapter.start: Starting connlib")
       do {
+        // We can only the system's default resolvers before connlib starts, and then they'll
+        // be overwritten from the ones from connlib. So cache them here for getSystemDefaultResolvers
+        // retrieve them later.
```
Comment (ReactorScram) [inline swift/apple/FirezoneNetworkExtension/Adapter.swift]:
```suggestion
        // We can only get the system's default resolvers before connlib starts, and then they'll
        // be overwritten by the ones from connlib. So cache them here for getSystemDefaultResolvers
        // to retrieve them later.
```

File: swift/apple/FirezoneNetworkExtension/Adapter.swift [hunk 2]
```diff
@@ -332,6 +338,12 @@ extension Adapter {
       self.logger.log("Adapter.didReceivePathUpdate: Back online. Starting connlib.")
 
       do {
+        // We can only the system's default resolvers before connlib starts, and then they'll
+        // be overwritten from the ones from connlib. So cache them here for getSystemDefaultResolvers
+        // retrieve them later.
```
Comment (ReactorScram) [inline swift/apple/FirezoneNetworkExtension/Adapter.swift]:
```suggestion
        // We can only get the system's default resolvers before connlib starts, and then they'll
        // be overwritten by the ones from connlib. So cache them here for getSystemDefaultResolvers
        // to retrieve them later.
```

**Review by ReactorScram — APPROVED** [review]
I skimmed through it.

So this Vec that caches the resolvers will move out of connlib and into the Apple client so that on other platforms, connlib can always retrieve the system resolvers, right?

**Comment by jamilbk** [comment]
> I skimmed through it.
> 
> So this Vec that caches the resolvers will move out of connlib and into the Apple client so that on other platforms, connlib can always retrieve the system resolvers, right?

Correct!

**Pull request merged**